InventoryComponents/InventoryConsumablesComponent.gd

```diff
@@ -11,19 +12,23 @@ func _init() -> void:
 	_update_consumable_limit()
 	consumable_slot_update.connect(_update_consumable_limit)
 
+##adds a consumable to the first open slot
 func add_consumable(consumable : Consumable) -> void:
 	var i : int = 0
 	
+	##This sycles through the slots and inserts a consumable into the first open slot it can find
 	for consumable_slot in _held_consumables:
 		if(consumable_slot == null):
-			set_consumable(consumable, i)
+			_set_consumable(consumable, i)
 			break
 		i += 1
 
-func set_consumable(consumable : Consumable, pos : int) -> void:
+##sets a given slot to the given consumable
+func _set_consumable(consumable : Consumable, pos : int) -> void:
 	_held_consumables[pos] = consumable
 	held_consumable_update.emit(consumable, pos)
 
+##empties out the consumable in given slot
```

Comment:
Sets the array of consumable to null at the given position, effectively removing the consumable at this position